common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -921,6 +921,17 @@ public static int collationNameToId(String collationName) throws SparkException
     return Collation.CollationSpec.collationNameToId(collationName);
   }
 
+  /**
+   * Returns whether the ICU collation is not Case Sensitive Accent Insensitive
+   * for the given collation id.
+   */
+  public static Boolean isNonCSAI(int collationId) {
+    return Collation.CollationSpecICU.fromCollationId(collationId).caseSensitivity ==
+            Collation.CollationSpecICU.CaseSensitivity.CI ||
+            Collation.CollationSpecICU.fromCollationId(collationId).accentSensitivity !=
+            Collation.CollationSpecICU.AccentSensitivity.AI;
+  }
```

Comment:
let's do `public static boolean` instead of `public static Boolean`

Reply:
also, I would prefer `isCaseSensitiveAccentInsensitive` instead of `isNonCSAI` here

Reply:
so something like:

```
  public static boolean isCaseSensitiveAccentInsensitive(int collationId) {
    return Collation.CollationSpecICU.fromCollationId(collationId).caseSensitivity == CS &&
            Collation.CollationSpecICU.fromCollationId(collationId).accentSensitivity == AI;
  }
```